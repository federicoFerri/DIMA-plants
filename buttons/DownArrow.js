import React from 'react';
import { StyleSheet, Image, TouchableOpacity, SafeAreaView } from 'react-native';

const image_DownArrow = require('../assets/button_images/arrow_down.png');

{/*props required
  @onPress: if button is pressed, it launches an event like TouchableOpacity
*/}

class DownArrow extends React.Component {
    imagePressed = () => () => {
        this.props.onPress()
    }
    render() {
        return(
            <SafeAreaView>
                <TouchableOpacity 
                    activeOpacity={0.5}
                    onPress={this.imagePressed()}
                >
                    <Image 
                        style={stylesImage.downArrow}
                        source={image_DownArrow}
                        /> 
                </TouchableOpacity>
            </SafeAreaView>
        )
    }
  
}


const stylesImage = StyleSheet.create({
    downArrow: {
      width: 32,
      height: 27,
      resizeMode: 'contain',
      marginTop: 0
    },
  });

export default DownArrow;