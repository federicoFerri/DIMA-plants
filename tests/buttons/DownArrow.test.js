import React from 'react';
import renderer from 'react-test-renderer';

import DownArrow from '../../buttons/DownArrow'

describe('<DownArrow />', () => {
    test('renders correctly', () => {
        const tree = renderer.create(<DownArrow />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});