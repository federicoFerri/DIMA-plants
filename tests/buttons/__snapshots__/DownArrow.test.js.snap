// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DownArrow /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "right": 5,
      "top": 10,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 27,
          "marginTop": 0,
          "resizeMode": "contain",
          "width": 32,
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;
